distributed/worker.py

```diff
@@ -1504,9 +1504,22 @@ async def start(self):
 
         setproctitle("dask-worker [%s]" % self.address)
 
-        await asyncio.gather(
-            *(self.plugin_add(plugin=plugin) for plugin in self._pending_plugins)
+        plugins_exceptions = await asyncio.gather(
+            *(
+                self.plugin_add(plugin=plugin, catch_errors=False)
+                for plugin in self._pending_plugins
+            ),
+            return_exceptions=True,
         )
+        if len(plugins_exceptions) >= 1:
```

Comment:
I think `plugins_exceptions` is not just the exceptions; it contains all the values returned by `plugin_add`, including OK messages. I think you'd want `plugins_exceptions = [x for x in plugins_results if isinstance(x, Exception)]` first.

Reply:
I don't think so, that was the case with `plugins_msgs` (the previous pickled version), but now adding a line just above this `print(f"plugins_exceptions: {plugins_exceptions}")`, I see the following for `distributed/tests/test_worker.py::test_plugin_exception`:

```
plugins_exceptions: [ValueError('Setup failed')]
```

and for `distributed/tests/test_worker.py::test_plugin_internal_exception` I see:

```
plugins_exceptions: [UnicodeDecodeError('utf-8', b'orrupting pickle\x80\x04\x95e\x02\x00\x00\x00\x00\x00\x00\x8c\x17cloudpickle.cloudpickle\x94\x8c\r_builtin_type\x94\x93\x94\x8c', 16, 17, 'invalid start byte')]
```

Reply:
Try having the plugin _not_ raise an exception and see what happens.

Reply:
Yes, you're absolutely right, thanks for pointing it out. This is fixed now by https://github.com/dask/distributed/pull/4298/commits/dc3aaeb6932103f4b29df3ee0b6ad956b98eb48f. Could you take another look when you have a chance?